libsolidity/ast/AST.cpp

```diff
@@ -480,6 +480,66 @@ bool VariableDeclaration::canHaveAutoType() const
 	return isLocalVariable() && !isCallableParameter();
 }
 
+bool VariableDeclaration::isEventParameter() const
+{
+	return dynamic_cast<EventDefinition const*>(scope()) != nullptr;
+}
+
+bool VariableDeclaration::isReferenceType() const
+{
+	return dynamic_cast<ReferenceType const*>(typeName()->annotation().type.get());
+}
+
+set<VariableDeclaration::Location> VariableDeclaration::getAllowedStorageLocations() const
+{
+	using Location = VariableDeclaration::Location;
+
+	if (isReferenceType())
```

Comment:
Could you invert this condition and return early to reduce the indentation level of the code below?